src/main/java/com/google/sps/servlets/AddSchoolData.java

```diff
@@ -0,0 +1,132 @@
+  private Key existingProfessorKey;
+
+  private DatastoreService db = DatastoreServiceFactory.getDatastoreService();
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    addSchoolData(db, request);
+    response.setContentType("text/html; charset=UTF-8");
+    response.setCharacterEncoding("UTF-8");
+    response.sendRedirect("/AddSchoolData.html");
+  }
+
+  public List<Entity> findQueryMatch(
+      String entityType, String entityProperty, String propertyValue) {
+    Filter filter = new FilterPredicate(entityProperty, FilterOperator.EQUAL, propertyValue);
+    Query q = new Query(entityType).setFilter(filter);
+    List<Entity> result = db.prepare(q).asList(FetchOptions.Builder.withDefaults());
+    return result;
+  }
+
+  public void isNewEntityDetector(String schoolName, String courseName, String profName) {
+    List<Entity> querySchool = findQueryMatch("School", "school-name", schoolName);
+    if (!querySchool.isEmpty()) {
+      existingSchoolKey = querySchool.get(0).getKey();
+      isNewSchool = false;
+    }
+
+    List<Entity> queryCourse = findQueryMatch("Course", "course-name", courseName);
+    if (!queryCourse.isEmpty()) {
+      existingSchoolKey = queryCourse.get(0).getKey();
+      isNewCourse = false;
+    }
+
+    List<Entity> queryProfessor = findQueryMatch("Professor", "professor-name", profName);
+    if (!queryProfessor.isEmpty()) {
+      existingSchoolKey = queryProfessor.get(0).getKey();
+      isNewProfessor = false;
+    }
+  }
+
+  public Entity createSchool(String schoolName) {
+    Entity newSchool = new Entity("School");
+    newSchool.setProperty("school-name", schoolName);
+    db.put(newSchool);
+    return newSchool;
+  }
+
+  public Entity createCourse(String name, Long units, Key parent) {
+    Entity newCourse = new Entity("Course", parent);
+    newCourse.setProperty("course-name", name);
+    newCourse.setProperty("units", units);
+    db.put(newCourse);
+    return newCourse;
+  }
+
+  public Entity createProfessor(String name, Key parent) {
+    Entity newProfessor = new Entity("Professor", parent);
+    newProfessor.setProperty("professor-name", name);
+    db.put(newProfessor);
+    return newProfessor;
+  }
+
+  public Entity createTerm(String term, Key professor, Key parent) {
+    Entity newTerm = new Entity("Term", parent);
+    newTerm.setProperty("term", term);
+    newTerm.setProperty("professorKey", professor);
+    db.put(newTerm);
+    return newTerm;
+  }
+
+  public void addSchoolData(DatastoreService db, HttpServletRequest request) {
+    String schoolName = request.getParameter("school-name");
+    String courseName = request.getParameter("course-name");
+    String termName = request.getParameter("term");
+    String profName = request.getParameter("professor-name");
+    Long units = Long.parseLong(request.getParameter("units"));
+
+    isNewEntityDetector(schoolName, courseName, profName);
```

Comment:
I'd turn this into three small functions that return booleans. That way you don't have to use global variables.